.changes/unreleased/NEW FEATURES-20250123-103202.yaml

```diff
@@ -0,0 +1,5 @@
+kind: NEW FEATURES
+body: 'tfdiags: Add Errors and Warnings methods to filter by severity'
```

Comment:
Given this is an internal refactoring and not user-facing change I think we should avoid a changelog entry.